docker/package/package_generator.py

```diff
@@ -2,29 +2,34 @@
 #
 # SPDX-License-Identifier: LicenseRef-MIT-TQ
 
-import os, shutil, sys, subprocess, json
+import os, shutil, sys, subprocess, json, argparse
 from distutils.dir_util import copy_tree
 
 from .model import Package, print_service_file
 from .packages import packages
 
-if sys.argv[1] == "ubuntu":
-    is_ubuntu = True
-elif sys.argv[1] == "fedora":
-    is_ubuntu = False
-else:
-    raise Exception(f"Unexpected package target OS, only 'ubuntu' and 'fedora' are supported.")
+is_ubuntu = None
+is_source = None
+package_to_build = None
+source_archive = None
 
-if sys.argv[2] == "source":
-    is_source = True
-elif sys.argv[2] == "binary":
-    is_source = False
-else:
-    raise Exception("Unexpected package format, only 'source' and 'binary' are supported.")
+parser = argparse.ArgumentParser()
+parser.add_argument("--os", default="ubuntu")
+parser.add_argument("--type", help="package type")
+parser.add_argument("--package", help="specify binary to package")
+parser.add_argument("--sources", help="specify source archive for single ubuntu package")
+args = parser.parse_args()
 
-package_to_build = None
-if len(sys.argv) == 4:
-    package_to_build = sys.argv[3]
+is_ubuntu = args.os if args.os is None else args.os == "ubuntu"
```

Comment:
AFAIU `is None` case is impossible because you provide a `default` value. You can also set `required=True` to be even more sure. And please set it for `--type` if it's required (instead of checking manually).

Reply:
ah, crap, I forgot to remove the default value for `--os`:(